app/backend/src/couchers/servicers/search.py

```diff
@@ -338,6 +345,196 @@ def _search_clusters(
     ]
 
 
+def _user_search_inner(request, context, session):
+    user = session.execute(select(User).where(User.id == context.user_id)).scalar_one()
+
+    # Base statement with visibility filter
+    statement = select(User.id).where_users_visible(context)
+
+    # If exactly_user_ids is present, only filter by those IDs and ignore all other filters
+    # This is a bit of a hacky feature to help with the frontend map implementation
+    if len(request.exactly_user_ids) > 0:
+        statement = statement.where(User.id.in_(request.exactly_user_ids))
+    else:
+        # Apply all the normal filters
+        if request.HasField("query"):
+            if request.query_name_only:
+                statement = statement.where(
+                    or_(User.name.ilike(f"%{request.query.value}%"), User.username.ilike(f"%{request.query.value}%"))
+                )
```

Comment:
I don't think we give an option on the frontend to say nameOnly for the search?

Reply:
There's a bunch of unused filters implemented in the backend.

I didn't modify this bit.